lib/Optimizer/Transforms/ObserveAnsatz.cpp

```diff
@@ -187,12 +171,30 @@ struct AppendMeasurements : public OpRewritePattern<func::FuncOp> {
       return failure();
     }
 
-    if (nMeasures != 0) {
-      std::string msg = "Cannot observe kernel with measures in it.\n";
-      funcOp.emitError(msg);
-      return failure();
+    // Attempt to remove measurements. Note that the mapping pass may add
```

Comment:
I'm not as familiar as I should be with the Mapping pass, but I think that pass will always tag the kernel with a specific attribute (the reorder one, right?). If this is true, it may be smart to also check that this attribute exists here, and throw a separate error if it does not exist but there are measurements. In this case the user of this pass has violated the specification of `cudaq::observe()` and used a kernel with measurements.

Reply:
I think it would be good to catch someone applying the observe ansatz pass from the command line on a kernel with measurements (apart from any physical topology mapping). 

I think adding a new simple pass that adds implicit measurements is a fine solution for the problems with Quantinuum. I wouldn't add it at the `nvq++` level, but more at the `RemoteRESTQPU` level. Just make sure they are ignorable in the observe ansatz pass.

Reply:
Updated in 8a9f69f96